### ISSUE TITLE
num_inference_timesteps can't exceed num_train_timesteps?

### ISSUE THREAD

**Issue by mihail911** [huggingface/diffusers#1005]
### Describe the bug

Hey all, 

I recently trained an unconditional image generation DDPM pipeline on a custom dataset using the `train_unconditional.py` script. The scheduler specified during training had `num_train_timesteps=1000` as follows:
```
noise_scheduler = DDPMScheduler(num_train_timesteps=1000)
```

Once I have trained the model and I am running inference, I try setting `num_inference_steps > 1000` in the DDPM pipeline as follows:
```
images = pipeline(
            batch_size=1,
            generator=torch.manual_seed(42),
            num_inference_steps=2000
        ).images
```

However the pipeline always only runs for 1000 timesteps. Is that to be expected? It's not clear to me why you can't denoise for more steps than you trained on here. 

Any help would be appreciated!

### System Info

- `diffusers` version: 0.7.0.dev0
- Platform: Linux-5.4.0-131-generic-x86_64-with-glibc2.27
- Python version: 3.9.13
- PyTorch version (GPU?): 1.12.1+cu102 (True)
- Huggingface_hub version: 0.10.1
- Transformers version: 4.23.1
- Using GPU in script?: YES
- Using distributed or parallel set-up in script?: NO

**Comment by patrickvonplaten** [comment]
Hey @mihail911,


At the moment one can indeed not set the number of inference steps higher than the number of training steps for **discrete schedulers**. We could make changes to allow for this, but it wouldn't make much sense - if the model has only been trained to handle steps for 0-1000 then it won't be able to correct understand steps above 1000.
Therefore I don't think denoising for more steps than were trained would lead to any better results

**Comment by mihail911** [comment]
Hey @patrickvonplaten - thanks for the response! I figured that was probably the reasoning, but just wanted to clarify if there was any harder constraint preventing this in the implementation. Either way, what you said makes sense.
